lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -345,23 +356,51 @@ def run
       unless config[:disable_customization]
         # Wait for customization to complete
         puts 'Waiting for customization to complete...'
-        CustomizationHelper.wait_for_sysprep(vm, vim, Integer(get_config(:sysprep_timeout)), 10)
+        CustomizationHelper.wait_for_sysprep(@vm, vim, Integer(get_config(:sysprep_timeout)), 10)
         puts 'Customization Complete'
-        sleep 2 until vm.guest.ipAddress
-        connect_host = config[:fqdn] = config[:fqdn] ? get_config(:fqdn) : vm.guest.ipAddress
+        sleep 2 until connect_host
```

Comment:
This sleep may be unnecessary as we sleep in the connect_host and ipv4_address methods.